src/main/java/com/ldtteam/structurize/config/AbstractConfiguration.java

```diff
@@ -91,11 +107,76 @@ protected static <T> ConfigValue<List<? extends T>> defineList(
         final List<? extends T> defaultValue,
         final Predicate<Object> elementValidator)
     {
-        return buildBase(builder, key).defineList(key, defaultValue, elementValidator);
+        return buildBase(builder, key, Component.empty()).defineList(key, defaultValue, elementValidator);
     }
 
     protected static <V extends Enum<V>> EnumValue<V> defineEnum(final Builder builder, final String key, final V defaultValue)
     {
-        return buildBase(builder, key).defineEnum(key, defaultValue);
+        return buildBase(builder,
+            key,
+            IcuComponent.of(DEFAULT_KEY_PREFIX + "enum",
+                defaultValue,
+                Arrays.stream(defaultValue.getDeclaringClass().getEnumConstants()).map(Enum::name).collect(Collectors.joining(", "))))
+            .defineEnum(key, defaultValue);
+    }
+
+    protected <T> void addWatcher(final ConfigValue<T> configValue, final ConfigListener<T> listener)
```

Comment:
Why do we have this odd watcher addition? Forge configs already have reloading on change, and an event that fires when it does. Also includes value caching

Reply:
yeah, but the event is fired for everything, you have no way to attach calls for static instances, eg. <https://github.com/ldtteam/Structurize/pull/613/files#diff-32d1c580005cc8075e20d9ae1ba1eff01b329e665837933039b6c76589c12224R42>
or i didnt find a way to do it

Reply:
not sure I understand why you need to attach a watcher instead of using ModConfigEvent.Reloading?

Reply:
repeating myself:
> edit: plus the mentioned event drives this watcher logic

this watcher is not a file watcher, its just class for checking if the value returned by forge config has changed, the event has no info about what was changed or what not, the only thing it has is the file that got changed, not which values or even if they actually have changed

Reply:
why do you need to know if a specific value changed? just reload everything

Reply:
well, im not gonna reconstruct our entire blueprint gpu cache just because irrelevant config has changed, if it would be off-thread then mby fine, but renderer is not there yet..